### PR TITLE
RavenDB-20372: Memory leak issue in RequestExecutor (v5.2)

### PR DESCRIPTION
**Pull request by ArieSLV**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-20372/Memory-leak-issue-in-RequestExecutor

### Additional description

Introduced the `Interlocked.Exchange` method to safely replace the old cluster request executor with the new one and dispose of the old one, ensuring that the memory leak issue is properly addressed.

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing by Contributor

- It has been verified by manual testing

### Testing by RavenDB QA team

- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Server/ServerWide/ServerStore.cs [hunk 1]
```diff
@@ -2983,8 +2983,10 @@ private async Task<(long Index, object Result)> SendToNodeAsync(TransactionOpera
                 || serverCertificateChanged
                 || _clusterRequestExecutor.Url.Equals(leaderUrl, StringComparison.OrdinalIgnoreCase) == false)
             {
-                _clusterRequestExecutor?.Dispose();
-                _clusterRequestExecutor = CreateNewClusterRequestExecutor(leaderUrl);
+                var newExecutor = CreateNewClusterRequestExecutor(leaderUrl);
+                var oldExecutor = Interlocked.Exchange(ref _clusterRequestExecutor, newExecutor);
+
+                oldExecutor?.Dispose();
```
Comment (ayende) [inline src/Raven.Server/ServerWide/ServerStore.cs:2989]: What happens if another thread is making use of the old executer while you dispose it? Shouldn't we worry about use after free here?

Comment (ArieSLV) [reply src/Raven.Server/ServerWide/ServerStore.cs:2989]: Oren, indeed, I need to ensure that no other threads are accessing the old executor before I dispose of it. I can implement a reference counting mechanism, incrementing the reference count when a thread begins using the executor and decrementing it once the thread has finished. The old executor would only be disposed of when the reference count reaches zero. Is this approach appropriate?

Comment (ppekrol) [reply src/Raven.Server/ServerWide/ServerStore.cs:2989]: Better to handle this in destructor?

Comment (ArieSLV) [reply src/Raven.Server/ServerWide/ServerStore.cs:2989]: Handling of the use-after-free issues implemented at https://github.com/ravendb/ravendb/pull/16345

**Pull request merged**